### PR TITLE
raylib: Add some useful headers to res folder

### PR DESCRIPTION
**Pull request by AbrilRBS**
### Summary
Changes to recipe:  **raylib/[*]**

#### Motivation

As explained in https://github.com/conan-io/conan-center-index/issues/24368, the library does not package some useful files that users might want to use as part of their build. This is similar to the imgui situation, so the solution is also similar: Add those files to the res folder, where users can then copy them from

#### Details

The linked issue and subsequent upstream issue have all the relevant info

As a draft until I get a chance to update the hook that will fail

cc @Ezbob

### CODE REVIEW

**Comment by Ezbob** [comment]
Hey @AbrilRBS , big thanks for this. 
I have a noob question though, how do you use/copy these headers in your conan file from a consumers perspective?

File: recipes/raylib/all/conanfile.py [hunk 1]
```diff
@@ -127,6 +134,9 @@ def package_info(self):
         elif self.settings.os == "Windows":
             self.cpp_info.system_libs.append("winmm")
 
+        # Some useful files are not packaged by default
+        self.cpp_info.resdirs = ["res/include"]
```
Comment (uilianries) [inline recipes/raylib/all/conanfile.py:138]:
```suggestion
        self.cpp_info.resdirs = ["res/include"]
        if self.conf.get("user.raylib:include_res", , default=False, check_type=bool):
            self.cpp_info.includedirs.append("res/include")
```

So users can use res folder directly without affecting the package ID.

**Comment by AbrilRBS** [comment]
Hi @Ezbob, not a noob question at all, I should have been more clear as to how to proceed, sorry about that!

We're going to make this available thru a conf as @uilianries suggested - I'll post a how-to after I commit the changes :)

**Comment by AbrilRBS** [comment]
With the current approach (pending dismissal from someone on the team if they don't like it) would be for you to compile your package as normal, but add the `user.raylib:include_res=True` conf, either in your CLI with `-c="user.raylib:include_res=True"`, or in your profile:
```
[conf]
user.raylib:include_res=True
```

This will make it so the recipe adds the new include path to your project, giving you access to those files.

The alternative approach if you don't need everything, or only need certain files, would be NOT to define the conf, and instead in your recipe's `generate()` method, have:

```py
def generate(self):
    # ... The rest of your generate method ...
    copy(self, pattern="rcamera.h", src=self.dependencies["raylib"].cpp_info.resdirs[0], dst=os.path.join(self.build_folder, "path-where-you-want-thefiles")
```

This would only copy rcamera.h and would give you the liberty of moving it to a place of your liking - hope this helps :)

**Comment by Ezbob** [comment]
@AbrilRBS Oh that look nice! Huge thanks for this!

**Comment by AbrilRBS** [comment]
Superseeded by https://github.com/conan-io/conan-center-index/pull/24585

**Pull request closed without merging**